core/src/test/java/org/elasticsearch/common/settings/SettingTests.java

```diff
@@ -154,6 +154,20 @@ public void testUpdaterIsIsolated() {
         assertNull(ab2.get());
     }
 
+    public void testDeprecatedSetting() {
```

Comment:
The reason I asked is you create a set from a CHM. Insertions and reads from the map can be concurrent as well as multiple calls that try to insert it into the set (caused by multiple Setting#get calls). I think the test could fire off concurrent gets of the setting to still validate that the message is only seen once.